File: Android/shared/src/main/kotlin/shared/dto/objects/HandleInvite.kt
@file:UseSerializers(
    UUIDSerializer::class,
    URISerializer::class
)
package shared.dto.objects

import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import shared.dto.protocols.Parametable
import shared.dto.protocols.Responsable
import shared.serializers.URISerializer
import shared.serializers.UUIDSerializer
import java.net.URI
import java.util.*

object HandleInvite {
    data object Parameters {
        @Serializable
        data class ClientMerge(
            val contact: String,
        ) : Parametable()
    }

    data object Responses {
        @Serializable
        data class EmployeeSuccess(
            val id: UUID,
            val nickname: String,
            val logo: URI,
            val position : String,
        ) : Responsable

        @Serializable
        data class MaskedContacts(
            val contacts : List<String>,
        ) : Responsable

        @Serializable
        data class ClientMergeSuccess(
            val id: UUID,
            val alias: String? = null,
        ) : Responsable
    }
}

File: iOS-legacy/DTOs/Objects/Common/HandleInvite.swift
import Foundation

public enum HandleInvite {
    public enum Parameters {}
    public enum Responses {}
    public enum Internal {}
}

public extension HandleInvite.Parameters {
  
    struct ClientMerge: Codable, Sendable {
        public var contact: String
        
        public init(contact: String) {
            self.contact = contact
        }
    }
}

public extension HandleInvite.Responses {
    
    struct EmployeeSuccess: Codable, Sendable {
        public let id: UUID
        public let nickname: String
        public let logo: URL
        public let position: String
        
        public init(id: UUID, nickname: String, logo: URL, position: String) {
            self.id = id
            self.nickname = nickname
            self.logo = logo
            self.position = position
        }
    }
    
    struct MaskedContacts: Codable, Sendable {
        public let contacts: [String]
        
        public init(contacts: [String]) {
            self.contacts = contacts
        }
    }
    
    struct ClientMergeSuccess: Codable, Sendable {
        public let id: UUID
        public let alias: String?
        
        public init(id: UUID, alias: String? = nil) {
            self.id = id
            self.alias = alias
        }
    }
}
